[Ryan Ong] IP

ryanongra opened the pull request:
# Do Understand Know Empower (Duke)

Hello 😄 This is Ryan Ong's evolution of the **Duke Chatbot** with the purpose of keeping track of the tasks *you* ~~die die~~ need to do.
> Quidquid latine dictum sit, altum videtur

## Features

 * Add tasks
 * Delete tasks
 * Complete tasks
 * Local storage

## Tasks

Tasks are available in the following types:

1. ToDo
2. Event
3. Deadline

## Example code

Here's an example of the `code` for the ToDo task!

```java
public static class ToDo extends Task {

        public ToDo(String name) throws DukeException.NoNameException {
            super(name, "#ToDo");
        }

        public ToDo(String name, boolean done) throws DukeException.NoNameException {
            super(name, "#ToDo", done);
        }
    }
```

## Have you tried all the features?

 - [x] Added three tasks (one of each type)
 - [x] Completed task 2
 - [ ] Deleted task 1
 - [ ] Closed the program
 - [ ] Restarted the program to see your list is saved!

## Guess what page I'm referring to now?

[This page!](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

ryanongra committed:
Update Duke.java

ryanongra committed:
Update Duke.java

ryanongra committed:
Update Duke.java

ryanongra committed:
Level-5

ryanongra committed:
Update Duke.java

ryanongra committed:
A-TextUiTesting

ryanongra committed:
Level-7

ryanongra committed:
Level-8

ryanongra committed:
Merge pull request #1 from ryanongra/branch-Level-7

Level-7

ryanongra committed:
Merge branch 'master' into branch-Level-8

ryanongra committed:
Merge pull request #2 from ryanongra/branch-Level-8

Level-8

ryanongra committed:
Clean up merge of branch-Level-7 and branch-Level-8

ryanongra committed:
A-MoreOOP

ryanongra committed:
A-Packages

ryanongra committed:
A-JUnit

ryanongra committed:
A-JavaDoc

ryanongra committed:
A-CodingStandard

ryanongra committed:
Level-9

ryanongra committed:
Update save.txt

ryanongra committed:
Merge pull request #3 from ryanongra/branch-A-JavaDoc

Merge branch-A-JavaDoc

ryanongra committed:
Merge pull request #4 from ryanongra/branch-A-CodingStandard

Merge branch-A-CodingStandard

ryanongra committed:
Merge pull request #5 from ryanongra/branch-Level-9

Merge branch-Level-9

cheweejia commented on src/main/java/Duke/ByeCommand.java:
```diff
@@ -0,0 +1,20 @@
+public class ByeCommand implements ICommand {
+
+    /**
+     * Called the Ui object to display the 'bye' message then ends the program.
```
Maybe it would be better to use "Calls" instead of "Called" here?

cheweejia commented on src/main/java/Duke/AddToDoCommand.java:
```diff
@@ -0,0 +1,34 @@
+ */
+public class AddToDoCommand implements ICommand {
+
+    private final String INPUT;
```
Should this be in lowercase since it is not really a constant but a class field?

cheweejia commented on src/main/java/Duke/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * Constructor for the command.
+     *
```
Just something very minor, but it may look nicer if you remove this empty line to standardise it with the javadocs for your other classes' constructors?

ryanongra committed:
Update Description

Update Description to use GFMD

cheweejia commented on src/main/java/Duke/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * Deletes the desired task by interacting with the relevant instances as mentioned above.
+     *
```
Same comment regarding the empty line for this part of the code too.

cheweejia commented on src/main/java/Duke/DukeException.java:
```diff
@@ -0,0 +1,38 @@
+package Duke;
+
+/**
+ * This class encapsulates exceptions which the Duke program could throw/
```
I think there is a typo at the end of this sentence?

cheweejia commented on src/main/java/Duke/FindTasksCommand.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class FindTasksCommand implements ICommand {
```
Maybe it would be good to include javadocs for the class and class methods too?

cheweejia commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,55 @@
+        } else if (input.equals("list")) {
+            return new ListTasksCommand();
+
+        } else if (input.split(" ")[0].equals("done")) {
```
Should `input.split(" ")[0]` be extracted out and assigned to a constant? This is so that the program does not have to carry out the split function every time it evaluates a conditional expression.

cheweejia commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    /**
+     * This method updates the local save file with the latest
```
Perhaps you should start with a verb like "Updates..."?

cheweejia commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,128 @@
+        this.taskType = taskType;
+    }
+
+    public String getName() {
```
Perhaps it will be better if there are javadocs for the public class and methods?

cheweejia commented on src/main/java/Duke/TaskManager.java:
```diff
@@ -0,0 +1,93 @@
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class TaskManager {
```
Perhaps it will be good to have javadocs for the class and class methods?

cheweejia reviewed (commented):
LGTM, just a few nits to fix.

ryanongra committed:
Merge pull request #6 from ryanongra/add-gradle-support

Add Gradle support

ryanongra committed:
Set up Gradle

ryanongra committed:
Set up checkstyle

czhi-bin commented on src/main/java/Duke/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+package Duke;
```
Perhaps you could group the class related to ICommand under one package like duke.command? It might help the reader to find the respective class?

czhi-bin commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,128 @@
+
+    public String toStringSave() {
+        int doneData = isDone ? 1 : 0;
+        char taskTypeData = taskType.equals("#ToDo")
+                ? 'T'
+                : taskType.equals("#Deadline")
+                ? 'D'
+                : 'E';
```
Perhaps you could add a bracket for the second ternary operator to make it less confusing?

czhi-bin reviewed (commented):
I think the code quality looks quite good! Just that perhaps you could package related class into the respective package to make it look more logical? Also, i think JavaDoc would be helpful for people to understand the usages of those Public classes and method?

ryanongra committed:
Add Duke GUI

ryanongra committed:
Refine GUI Integration

ryanongra committed:
Merge pull request #7 from ryanongra/branch-Level-10

Refine GUI Integration

yewmeng28 commented on README.md:
````diff
@@ -1,24 +1,48 @@
+```java
+public static class ToDo extends Task {
+
+        public ToDo(String name) throws DukeException.NoNameException {
````
The indentation of these methods seems a little too far to the right.

yewmeng28 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                String name = taskDetails[2];
+                boolean done = taskDetails[1].equals("1");
+                switch (taskDetails[0]) {
+                    case "T":
```
The `case` in `switch` statements should be at the same level of indentation as the `switch` itself. You can refer to the module dashboard or online on how to change this in IntelliJ.

yewmeng28 commented on src/main/java/duke/AddToDoCommand.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    public void execute(TaskManager tm, ResponseManager responseManager, Storage storage) {
+        try {
+            Task addedTask = tm.addToDo(input);
```
Perhaps this can be simplified to `task`.

yewmeng28 reviewed (commented):
Just some minor tweaks to your code, other than that it looks good!

ryanongra committed:
A-Assertions

ryanongra committed:
Merge pull request #8 from ryanongra/branch-A-Assertions

Merge branch-A-Assertions

ryanongra committed:
Implement A-CodeQuality

Ensure code quality based on the coding standards and conventions for this module.

Major amends include:
 - Ensuring that each class has JavaDocs for their constructors minimally.
 - Each local variable has access modifiers.
 - JavaDocs format standardised based on coding convention.

This was done using the checkstyle module adding from a previous commit.
 -

ryanongra committed:
Merge branch 'master' into branch-A-CodeQuality

ryanongra committed:
Merge pull request #9 from ryanongra/branch-A-CodeQuality

Implement A-CodeQuality

ryanongra committed:
Set up CI

ryanongra committed:
Merge pull request #10 from ryanongra/branch-A-CI

Set up CI

ryanongra committed:
Add Snooze function

The snooze function serves the same function as rescheduling a task. It will take a task index and a date and time as an input and will change the time set for that task to the entered time.

The potential exceptions handled include:
 - Attempting to use snooze on a task which does not have a dateTime (ToDo task).
 - Calling snooze without providing a date.
 - Calling snooze on an invalid task index.

ryanongra committed:
Merge pull request #11 from ryanongra/branch-B-Snooze

Add Snooze function

ryanongra committed:
Add Dark Theme and Help button

ryanongra committed:
Merge pull request #13 from ryanongra/branch-A-BetterGui

Add Dark Theme and Help button

ryanongra committed:
Refactor additional packages

ryanongra committed:
Add screenshot

ryanongra committed:
Update Ui.png.png

ryanongra committed:
Update Ui.png

ryanongra committed:
Update README

ryanongra committed:
Update README.md

ryanongra committed:
Update README.md

ryanongra committed:
Update Help Button Link

Change the placeholder link for the help button to redirect the user to the user guide on github.

ryanongra committed:
Merge branch 'master' of https://github.com/ryanongra/ip

ryanongra committed:
Set theme jekyll-theme-merlot

ryanongra committed:
Set theme jekyll-theme-midnight

ryanongra committed:
Update README.md

ryanongra committed:
Update README.md

ryanongra committed:
Update README.md

ryanongra committed:
Update README.md

ryanongra committed:
Clean up code

Apply SLAP
Ensure coding standard
Add comments

ryanongra committed:
Merge branch 'master' of https://github.com/ryanongra/ip

ryanongra committed:
Add tests for ResponseManager

ryanongra committed:
Add tests

Add JUnit tests
Add assertions

ryanongra committed:
Update build.gradle

ryanongra committed:
Fix Jar File

ryanongra committed:
Fix Jar File
